dali/operators/generic/reshape.cc

```diff
@@ -219,9 +251,14 @@ void Reshape<Backend>::CalculateOutputShape(const Workspace &ws) {
       break;
     case ShapeSource::None:
       output_shape_ = input_shape_;
-      return;
+      break;
   }
 
+  int64_t input_element_size = in.type().size();
```

Comment:
a suggestion: You could check output_type_ earlier and later just assume that output_type_ is a valid type
```
if (output_type_ != DALI_NO_TYPE)
    output_type_ = in.type();
```

Reply:
Done.